File: resources/exercises/bat/bool2/AnswerCellEntity.java
/* automatically converted from the Nick Parlante's excellent exercising site http://javabat.com/ */

package bat.bool2;
import plm.core.model.lesson.ExerciseTemplated;
import plm.core.model.lesson.Lesson;
import plm.universe.bat.BatEntity;
import plm.universe.bat.BatTest;
import plm.universe.bat.BatWorld;

public class AnswerCellEntity extends BatEntity {
	public void run(BatTest t) {
		t.setResult( answerCell((Boolean)t.getParameter(0), (Boolean)t.getParameter(1), (Boolean)t.getParameter(2)) );
	}

	/* BEGIN TEMPLATE */
	boolean answerCell(boolean isMorning, boolean isMom, boolean isAsleep) {
		/* BEGIN SOLUTION */
		return (! isAsleep) && ! (isMorning && ! isMom);
		/* END SOLUTION */
	}
	/* END TEMPLATE */
}

File: src/lessons/welcome/bat/bool2/AnswerCellEntity.py
def run():
    count = getTestCount()
    for i in range(count):
        param = deserialize(getTest(i))
        setTestResult(i, serialize(answerCell(param[0], param[1], param[2])))

# BEGIN TEMPLATE
def answerCell(isMorning, isMom, isAsleep):
# BEGIN SOLUTION
   return (not isAsleep) and not (isMorning and not isMom)
# END SOLUTION
# END TEMPLATE
